chia/full_node/sync_store.py

```diff
@@ -75,11 +70,11 @@ def peer_has_block(
             if len(self.peak_to_peer) > 256:  # nice power of two
                 item = self.peak_to_peer.popitem(last=False)  # Remove the oldest entry
                 # sync target hash is used throughout the sync process and should not be deleted.
-                if item[0] == self.sync_target_header_hash:
+                if self.target_peak is not None and item[0] == self.target_peak.header_hash:
                     self.peak_to_peer[item[0]] = item[1]  # Put it back in if it was the sync target
                     self.peak_to_peer.popitem(last=False)  # Remove the oldest entry again
         if new_peak:
-            self.peer_to_peak[peer_id] = (header_hash, height, weight)
+            self.peer_to_peak[peer_id] = Peak(header_hash, height, weight)
```

Comment:
We could add an invariant here and throw if we expect the components of `Peak` (header_hash, height, weight) to be non-null. Or perhaps in the Peak constructor?

Reply:
Why do you think we need this here since they are supposed to be non-optional by their definition `header_hash: bytes32, peer_id: bytes32, weight: uint128, height: uint32`?

Reply:
We don't need it - it was just a suggestion. Since we enforce running mypy in the calling code, this is good enough.